validator_client/src/attestation_data_service.rs

```diff
@@ -0,0 +1,84 @@
+use std::{collections::HashMap, sync::Arc};
+
+use slot_clock::SlotClock;
+use types::{AttestationData, CommitteeIndex, EthSpec, ForkName, Slot};
+
+use crate::{
+    beacon_node_fallback::{BeaconNodeFallback, OfflineOnFailure, RequireSynced},
+    http_metrics::metrics,
+};
+
+/// The AttestationDataService is responsible for downloading and caching attestation data at a given slot
+/// for a range of committee indexes. It also helps prevent us from re-downloading identical attestation data.
+pub struct AttestationDataService<T: SlotClock, E: EthSpec> {
+    attestation_data_by_committee: HashMap<CommitteeIndex, AttestationData>,
```

Comment:
Note that all attestations votes are the same on each committee, the only change is in the `index` field which equals the key of this hashmap `CommitteeIndex`. If you are refactoring the flow you can consider an optimization to only get a single `AttestationData` for all committees.

However, some DVT solution relies on a call for each `CommitteeIndex` being made. I'm not sure if this is the case but it was ~1.5 years ago.